src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -94,6 +94,29 @@ FixedwingPositionControl::init()
 	return true;
 }
 
+float FixedwingPositionControl::getMaximumClimbRate()
+{
+
+	float weight_factor = 1.0f;
+
+	if (_param_weight_base.get() > 0.0f && _param_weight_gross.get() > 0.0f) {
+		weight_factor = math::constrain(_param_weight_base.get() / _param_weight_gross.get(), MIN_WEIGHT_RATIO,
+						MAX_WEIGHT_RATIO);
+	}
+
+	float climbrate_max = _param_fw_t_clmb_max.get();
+
+	if (_param_density_min.get() > 0.0f) {
+		const float min_density = math::max(_param_density_min.get(), AIR_DENSITY_STANDARD_ATMOS_5000_AMSL);
```

Comment:
Either constrain to a reasonable max air density as well or skip the correct if it doesn't seem valid?